src/posix.mak

```diff
@@ -477,4 +477,14 @@ zip:
 
 #############################
 
+ifneq ($(DOCDIR),)
+html: $(DOCDIR)/.generated
+$(DOCDIR)/.generated: $(DMD_SRCS) $(ROOT_SRCS) $(HOST_DMD_PATH)
+	$(HOST_DMD_RUN) -of- $(MODEL_FLAG) -J. -c -Dd$(DOCDIR)\
+	  $(DFLAGS) $(DOCFMT) $(DMD_SRCS) $(ROOT_SRCS)
+	touch $@
+endif
```

Comment:
I would have provided an error message if the condition if false, especially since the rule is present at the top level.
This is what you get when you type `make html` at the top level:

```
% make -f posix.mak html                                                                                    :(
make -C src -f posix.mak html
make[1]: Entering directory '/mnt/shared/projects/dlang/dmd/src'
no cpu specified, assuming X86
make[1]: *** No rule to make target 'html'.  Stop.
make[1]: Leaving directory '/mnt/shared/projects/dlang/dmd/src'
posix.mak:24: recipe for target 'html' failed
make: *** [html] Error 2
```

It's not really informative of what went wrong.